projects/frontend/src/draft-posts/index.md

```diff
@@ -0,0 +1,148 @@
+
+## Simple Go Reverse Proxy
+How do we go about rolling our own simple proxy in go?
+
+```go
+proxy := httputil.NewSingleHostReverseProxy(url)
+```
+
+Yep, with the go standard library it really is that simple.
+
+Let’s dig in here. The function returns a reverse proxy with the following method.
+
+```go
+func (p *ReverseProxy) ServeHTTP(rw http.ResponseWriter, req *http.Request) {
+```
+
+All we need to do is configure the proxy and wire it up to a standard go http server to have a working reverse proxy as shown below.
+
+```go
+package main
+
+import (
+	"flag"
+	"log"
+	"net/http"
+	"net/http/httputil"
+	"net/url"
+	"strconv"
+)
+
+func main() {
+	port := flag.Int("port", 8080, "port to listen on")
+	targetURL := flag.String("target-url", "", "downstream service url to proxy to")
+	flag.Parse()
+
+	u, err := url.Parse(*targetURL)
+	if err != nil {
+		log.Fatalf("Could not parse downstream url: %s", *targetURL)
+	}
+
+	proxy := httputil.NewSingleHostReverseProxy(u)
+
+	director := proxy.Director
+	proxy.Director = func(req *http.Request) {
+		director(req)
+		req.Header.Set("X-Forwarded-Host", req.Header.Get("Host"))
+		req.Host = req.URL.Host
+	}
+
+	http.HandleFunc("/", proxy.ServeHTTP)
+	log.Printf("Listening on port %d", *port)
+	log.Fatal(http.ListenAndServe(":"+strconv.Itoa(*port), nil))
+}
+```
+
+That’s it! This server can now handle proxying HTTP 1 and 2 as well as web socket connections. You’ll notice that I’ve configured the _proxy.Director_ field. The _ReverseProxy.Director_ is a function that modifies the incoming request before it is forwarded. The signature is as follows:
+
+```go
+Director func(*http.Request)
+```
+
+A common use case for the director function is altering request headers. There is a default director set by  _httputil.NewSingleHostReverseProxy_ . The default implementation takes care of setting the request scheme, host, and path. I found this useful and didn’t want to duplicate the code. I had to reset the _req.Host_  field to contact HTTPS endpoints . I’ve also included an example of setting a request header.
+
+## Capturing Metrics 
+Not too long ago I was asked to build a demo for one of our clients at work. The task was to capture the response metrics of a web service and add those metrics to [Prometheus](https://prometheus.io/) for display in [Grafana](https://grafana.com/). The only catch was that I did not have direct access to the web service and therefore couldn’t instrument the code. I completed the task by adapting the go reverse proxy to observe the network traffic coming from the service.
```

Comment:
This is the explanation of why that should be started a lot earlier.

Also, this is the place where people typically plug their company 😃

Reply:
I just suggested you split this up and bring some of the "why" earlier, but if you don't, I'd suggest an edit of this section. It doesn't feel like it has a direct enough relationship to the previous section. Something along the lines of: 
```markdown
One way reverse proxies are useful is they allow you to insert instrumentation around an http service without having to modify the service itself.
``` 
followed by a slight tweak of your paragraph. Also, it's an implementation of the decorator pattern, but your HTTP service is the function!